File: mobile-app/app/src/main/java/com/example/healthbook/MainActivity.java
package com.example.healthbook;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;
import androidx.navigation.ui.NavigationUI;
import com.example.healthbook.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        // Seed data to Firestore if needed
        new com.example.healthbook.data.ApiRepository().seedDataIfNeeded();

        NavHostFragment navHostFragment = (NavHostFragment) getSupportFragmentManager()
                .findFragmentById(R.id.nav_host_fragment);
        
        if (navHostFragment != null) {
            NavController navController = navHostFragment.getNavController();
            NavigationUI.setupWithNavController(binding.navView, navController);

            navController.addOnDestinationChangedListener((controller, destination, arguments) -> {
                int id = destination.getId();
                if (id == R.id.navigation_home || id == R.id.navigation_appointments || 
                    id == R.id.navigation_profile || id == R.id.navigation_notifications || 
                    id == R.id.navigation_account || id == R.id.navigation_doctor_home ||
                    id == R.id.navigation_doctor_schedule || id == R.id.navigation_doctor_appointments ||
                    id == R.id.navigation_admin_dashboard || id == R.id.navigation_admin_users ||
                    id == R.id.navigation_admin_hospitals) {
                    binding.navView.setVisibility(android.view.View.VISIBLE);
                } else {
                    binding.navView.setVisibility(android.view.View.GONE);
                }
            });
        }
    }

    public void setupNavigationForRole(String role) {
        if (role == null) role = "patient";
        
        binding.navView.getMenu().clear();
        NavHostFragment navHostFragment = (NavHostFragment) getSupportFragmentManager()
                .findFragmentById(R.id.nav_host_fragment);
        
        if (navHostFragment == null) return;
        NavController navController = navHostFragment.getNavController();

        if (role.equalsIgnoreCase("admin")) {
            binding.navView.inflateMenu(R.menu.bottom_nav_menu_admin);
            navController.navigate(R.id.navigation_admin_dashboard);
        } else if (role.equalsIgnoreCase("doctor")) {
            binding.navView.inflateMenu(R.menu.bottom_nav_menu_doctor);
            navController.navigate(R.id.navigation_doctor_home);
        } else {
            binding.navView.inflateMenu(R.menu.bottom_nav_menu);
            navController.navigate(R.id.navigation_home);
        }
    }
}

File: mobile-app/app/src/main/java/com/example/healthbook/MainActivity.kt
package com.example.healthbook

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.example.healthbook.ui.theme.HealthBookTheme

import androidx.navigation.compose.rememberNavController
import com.example.healthbook.navigation.HealthBookNavGraph

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // enableEdgeToEdge()
        setContent {
            HealthBookTheme {
                val navController = rememberNavController()
                HealthBookNavGraph(navController = navController)
            }
        }
    }
}